go/oasis-node/cmd/debug/txsource/workload/commission.go

```diff
@@ -0,0 +1,251 @@
+		// In case we are setting more bound steps than rates, make sure that
+		// bounds after last rate step are not exclusive, as the last rate rule
+		// will need to be within all following bounds.
+		var maxBound, minBound int64
+		if i >= nRateSteps {
+			prevMaxBound := amendSchedule.Amendment.Bounds[i-1].RateMax.ToBigInt().Int64()
+			prevMinBound := amendSchedule.Amendment.Bounds[i-1].RateMin.ToBigInt().Int64()
+			// [prevMinBound, prevMaxBound]
+			maxBound = rng.Int63n(prevMaxBound-prevMinBound+1) + prevMinBound
+			// [prevMinBound, newMaxBound]
+			minBound = rng.Int63n(maxBound-prevMinBound+1) + prevMinBound
+		} else {
+			// [10, 100_000]
+			maxBound = rng.Int63n(100_000-10+1) + 10
+			// [0, maxBound]
+			minBound = rng.Int63n(maxBound + 1)
+		}
+		bound := staking.CommissionRateBoundStep{
+			Start: boundEpoch,
+		}
+		if err = bound.RateMin.FromInt64(minBound); err != nil {
+			return fmt.Errorf("Rate.FromInt64 err: %w", err)
+		}
+		if err = bound.RateMax.FromInt64(maxBound); err != nil {
+			return fmt.Errorf("Rate.FromInt64 err: %w", err)
+		}
+		amendSchedule.Amendment.Bounds = append(amendSchedule.Amendment.Bounds, bound)
+	}
+
+	// Genarete rates.
+	for i, rateEpoch := range rateStepEpochs {
+		// Get bounds for i'th epoch.
+		var bound staking.CommissionRateBoundStep
+		if i < nBoundSteps {
+			bound = amendSchedule.Amendment.Bounds[i]
+		} else {
+			// In case there are more rate steps than bounds, last bound will
+			// be in effect.
+			bound = amendSchedule.Amendment.Bounds[nBoundSteps-1]
+		}
+		minBound := bound.RateMin.ToBigInt().Int64()
+		maxBound := bound.RateMax.ToBigInt().Int64()
+		// [minBound, maxBound]
+		rateN := rng.Int63n(maxBound-minBound+1) + minBound
+		rate := staking.CommissionRateStep{Start: rateEpoch}
+		if err = rate.Rate.FromInt64(rateN); err != nil {
+			return fmt.Errorf("Rate.FromInt64 err: %w", err)
+		}
+		amendSchedule.Amendment.Rates = append(amendSchedule.Amendment.Rates, rate)
+	}
+
+	// In case there is more bounds than rate steps, fix the last rate step to
+	// be within all bounds that follow.
+	if nBoundSteps > nRateSteps {
+		// Find the greatest RateMin, and least RateMax in bounds that follow
+		// the last rate.
+		// NOTE: we make sure that these rates are not exclusive when generating
+		// the bounds.
+		var minBound int64
+		maxBound := int64(100_000)
+		for i := nRateSteps - 1; i < nBoundSteps; i++ {
+			boundMax := amendSchedule.Amendment.Bounds[i].RateMax.ToBigInt().Int64()
+			boundMin := amendSchedule.Amendment.Bounds[i].RateMin.ToBigInt().Int64()
+			if boundMax < maxBound {
+				maxBound = boundMax
+			}
+			if boundMin > minBound {
+				minBound = boundMin
+			}
+		}
+
+		// [minBound, maxBound]
+		rateN := rng.Int63n(maxBound-minBound+1) + minBound
+		if err = amendSchedule.Amendment.Rates[nRateSteps-1].Rate.FromInt64(rateN); err != nil {
+			return fmt.Errorf("Rate.FromInt64 err: %w", err)
+		}
+	}
+
+	// Generate and submit the transaction.
```

Comment:
a separate comment below says that we submit the transaction

Reply:
fixed